src/axom/inlet/Table.hpp

```diff
@@ -939,74 +817,145 @@ class Table : public Verifiable<Table>
    *****************************************************************************
    */
-  bool contains(const std::string& name) const;
+  Table& addTable(const std::string& name, const std::string& description = "");
 
   /*!
    *****************************************************************************
-   * \brief Returns whether this table or any of its subtables contain a non-
-   * empty field
+   * \brief Add an array of primitive Fields to the input file schema.
+   *
+   * \param [in] name Name of the array
+   * \param [in] description Description of the Field
+   * \param [in] isDict Whether to use string-valued keys for the container
+   * \param [in] pathOverride The path within the input file to read from, if
+   * different than the structure of the Sidre datastore
+   *
+   * \return Reference to the created Field
    *****************************************************************************
    */
-  explicit operator bool() const;
+  template <typename T,
+            typename SFINAE =
+              typename std::enable_if<detail::is_inlet_primitive<T>::value>::type>
+  Verifiable<Table>& addPrimitiveArray(const std::string& name,
+                                       const std::string& description = "",
+                                       const bool isDict = false,
+                                       const std::string& pathOverride = "");
 
   /*!
    *****************************************************************************
-   * \return An unordered map from Field names to the child Field pointers for 
-   * this Table.
+   * \brief Return whether a Table with the given name is present in this Table's subtree.
+   *
+   * \return Boolean value indicating whether this Table's subtree contains this Table.
    *****************************************************************************
    */
-  const std::unordered_map<std::string, std::unique_ptr<Field>>& getChildFields() const;
+  bool hasTable(const std::string& tableName) const;
 
   /*!
    *****************************************************************************
-   * \return An unordered map from Table names to the child Table pointers for 
-   * this Table.
+   * \brief Return whether a Field with the given name is present in this Table's
+   *  subtree.
+   *
+   * \return Boolean value indicating whether this Table's subtree contains this Field.
    *****************************************************************************
    */
-  const std::unordered_map<std::string, std::unique_ptr<Table>>& getChildTables() const;
+  bool hasField(const std::string& fieldName) const;
 
   /*!
    *****************************************************************************
-   * \return The full name of this Table.
+   * \brief Return whether a Function with the given name is present in this Table's
+   *  subtree.
+   *
+   * \return Boolean value indicating whether this Table's subtree contains this Function.
```

Comment:
too many this's.  possibly "this Table's subtree contains a function with the given name"